procs.rb

```diff
@@ -0,0 +1,134 @@
+# Use Blocks, Procs and Lambda each to write a program which takes array of integers as an input and finds the count of even numbers in the array
+# Method using block
+def count_even_numbers_block(arr)
+    count = 0
+    arr.each { |num| count += 1 if num.even? }
+    count
+  end
+
+  # Method using proc
+  def count_even_numbers_proc(arr, count_proc)
+    count_proc.call(arr)
+  end
+
+  # Method using lambda
+  def count_even_numbers_lambda(arr, count_lambda)
+    count_lambda.call(arr)
+  end
+
+
+  numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+
+  #block
+  count_using_block = count_even_numbers_block(numbers)
+  puts "Count of even numbers using block: #{count_using_block}"
+
+  #proc
+  count_proc = proc { |arr| arr.count(&:even?) }
+  count_using_proc = count_even_numbers_proc(numbers, count_proc)
+  puts "Count of even numbers using proc: #{count_using_proc}"
+
+  #lambda
+  count_lambda = ->(arr) { arr.count(&:even?) }
+  count_using_lambda = count_even_numbers_lambda(numbers, count_lambda)
+  puts "Count of even numbers using lambda: #{count_using_lambda}"
+
+#   Object Orientation
+#   Write a program in ruby which represents a single Company which has employees, departments and projects. 
+class Employee
+  attr_accessor :name
+
+  def initialize(name)
+    @name = name
+  end
+end
+
+class Department
+  attr_accessor :name, :department_head, :employees
+
+  def initialize(name, department_head)
+    @name = name
+    @department_head = department_head
+    @employees = []
+  end
+
+  def add_employee(employee)
+    @employees << employee
+  end
+
+  def employee_count
+    @employees.length
+  end
+end
+
+class Company
```

Comment:
please create separate files for each class. You shouldn't define them in single file. Each class should be defined in separate files.